components/eam/src/chemistry/modal_aero/seasalt_model.F90

```diff
@@ -475,68 +503,67 @@ subroutine seasalt_emis(u10, u10cubed, lchnk, srf_temp, ocnfrc, ncol, cflx, emis
-      mm = seasalt_indices(nslt+m_om)
+   om_mode_loop: do ispec=1,nslt_om
+      mass_mode_idx = seasalt_indices(nslt+ispec)
 
-      cflx(:ncol,mm)=0.0_r8
+      cflx(:ncol,mass_mode_idx)=0.0_r8
       
-      if (emit_this_mode(m_om)) then
+      if (emit_this_mode(ispec)) then
          ! write(iulog,"(A30,A10,I3)") "Constituent name and number: ", trim(seasalt_names(nslt+m_om)), mm ! for debugging
          ! add mass tracers
-         om_type_loop: do n=1,n_org
-            section_loop_OM_mass: do i=1, nsections
-               if (Dg(i)>=sst_sz_range_lo(nslt+m_om) .and. Dg(i)<sst_sz_range_hi(nslt+m_om)) then
-                  cflx_help2(:ncol)=fi(:ncol,i)*ocnfrc(:ncol)*emis_scale &
-                                    *4._r8/3._r8*pi*rdry(i)**3*dns_aer_sst  ! should use dry size, convert from number to mass flux (kg/m2/s)
+         om_type_loop: do iorg = 1, n_org
+            section_loop_OM_mass: do ibin = 1, nsections
+               if (Dg(ibin)>=sst_sz_range_lo(nslt+ispec) .and. Dg(ibin)<sst_sz_range_hi(nslt+ispec)) then
+                  cflx_tmp1(:ncol) = fi(:ncol,ibin)*ocnfrc(:ncol)*emis_scale &
+                                     *4._r8/3._r8*pi*rdry(ibin)**3*dns_aer_sst  ! should use dry size, convert from number to mass flux (kg/m2/s)
                   !  mass_frac_bub_section(pcols, n_org_max, nsections) -- org classes in dim 2, size nsections in dim 3
              
                   ! Mixing state 1: external mixture, add OM to mass and number
                   ! Mixing state 3: internal mixture, add OM to mass and number
-                  where (om_ssa(:ncol, i) > 0.0_r8) ! avoid division by zero
-                     cflx(:ncol,mm) = cflx(:ncol,mm) + cflx_help2(:ncol) &
-                                    * mass_frac_bub_section(:ncol, n, i) / om_ssa(:ncol, i) * &
-                                      (1._r8 / (1._r8 - om_ssa(:ncol, i)) - 1._r8)
+                  where (om_ssa(:ncol,ibin) > 0.0_r8) ! avoid division by zero
+                     cflx(:ncol,mass_mode_idx) = cflx(:ncol,mass_mode_idx) + cflx_tmp1(:ncol) &
+                                                 * mass_frac_bub_section(:ncol, iorg, ibin) / om_ssa(:ncol, ibin) * &
+                                                 (1._r8 / (1._r8 - om_ssa(:ncol, ibin)) - 1._r8)
                   elsewhere
-                     cflx(:ncol,mm) = cflx(:ncol,mm)
+                     cflx(:ncol,mass_mode_idx) = cflx(:ncol,mass_mode_idx)
                   endwhere
                endif
             enddo section_loop_OM_mass
          enddo om_type_loop
       endif
    enddo om_mode_loop
 
-end subroutine seasalt_emis
+end subroutine marine_organic_emis
 
 
-subroutine calc_om_ssa_burrows(ncol, mpoly_in, mprot_in, mlip_in, &
-                                mass_frac_bub_section, om_ssa, F_eff, lchnk)
+subroutine calc_om_ssa_burrows(lchnk, ncol, mpoly_in, mprot_in, mlip_in, &
+                               mass_frac_bub_section, om_ssa)
 
```

Comment:
Does lchnk still need to be passed to this subroutine?

Reply:
removed lchnk, renamed to calc_om_ssa